src/main.rs

```diff
@@ -0,0 +1,439 @@
+// SPDX-FileCopyrightText: 2020 Serokell <https://serokell.io/>
+//
+// SPDX-License-Identifier: MPL-2.0
+
+use clap::Clap;
+
+use std::process::Stdio;
+use tokio::process::Command;
+
+use merge::Merge;
+
+extern crate pretty_env_logger;
+
+#[macro_use]
+extern crate log;
+
+#[macro_use]
+extern crate serde_derive;
+
+#[macro_use]
+mod utils;
+
+/// Simple Rust rewrite of a simple Nix Flake deployment tool
+#[derive(Clap, Debug)]
+#[clap(version = "1.0", author = "Serokell <https://serokell.io/>")]
+struct Opts {
+    /// The flake to deploy
+    #[clap(default_value = ".")]
+    flake: String,
+    /// Check signatures when using `nix copy`
+    #[clap(short, long)]
+    checksigs: bool,
+    /// Extra arguments to be passed to nix build
+    extra_build_args: Vec<String>,
+
+    /// Keep the build outputs of each built profile
+    #[clap(short, long)]
+    keep_result: bool,
+    /// Location to keep outputs from built profiles in
+    #[clap(short, long)]
+    result_path: Option<String>,
+
+    /// Skip the automatic pre-build checks
+    #[clap(short, long)]
+    skip_checks: bool,
+
+    /// Override the SSH user with the given value
+    #[clap(long)]
+    ssh_user: Option<String>,
+    /// Override the profile user with the given value
+    #[clap(long)]
+    profile_user: Option<String>,
+    /// Override the SSH options used
+    #[clap(long)]
+    ssh_opts: Option<String>,
+    /// Override if the connecting to the target node should be considered fast
+    #[clap(long)]
+    fast_connection: Option<bool>,
+    /// Override if a rollback should be attempted if activation fails
+    #[clap(long)]
+    auto_rollback: Option<bool>,
+    /// Override hostname used for the node
+    #[clap(long)]
+    hostname: Option<String>,
+    /// Make activation wait for confirmation, or roll back after a period of time
+    #[clap(long)]
+    magic_rollback: Option<bool>,
+    /// How long activation should wait for confirmation (if using magic-rollback)
+    #[clap(long)]
+    confirm_timeout: Option<u16>,
+    /// Where to store temporary files (only used by magic-rollback)
+    #[clap(long)]
+    temp_path: Option<String>,
+}
+
+async fn push_all_profiles(
+    node: &utils::data::Node,
+    node_name: &str,
+    supports_flakes: bool,
+    repo: &str,
+    top_settings: &utils::data::GenericSettings,
+    check_sigs: bool,
+    cmd_overrides: &utils::CmdOverrides,
+    keep_result: bool,
+    result_path: Option<&str>,
+) -> Result<(), Box<dyn std::error::Error>> {
+    info!("Pushing all profiles for `{}`", node_name);
+
+    let mut profiles_list: Vec<&str> = node
+        .node_settings
+        .profiles_order
+        .iter()
+        .map(|x| x.as_ref())
+        .collect();
+
+    // Add any profiles which weren't in the provided order list
```

Comment:
This is specifically inside `push_all_profiles` which is only called if you run `deploy` or `deploy .#node`

Reply:
Well, yes. More documentation will solve this issue.

Reply:
Ah I think I missed what you meant, I've added a note about it now. The reason it will add all of them is because I can see a use case where only the order of some of the profiles would be important, the rest can run arbitrarily afterwards, though if limiting the profiles which are automatically deployed is desirable I can change this